bootstrap/handlers/httpcors.go

```diff
@@ -46,27 +50,35 @@ func ProcessCORS(corsInfo config.CORSConfigurationInfo) func(http.Handler) http.
 
 				w.Header().Set(Vary, Origin)
 			}
-
-			next.ServeHTTP(w, r)
-		})
+			return next(c)
+		}
 	}
 }
 
 // HandlePreflight returns a http handler function that process CORS preflight responses and sets CORS headers.
-func HandlePreflight(corsInfo config.CORSConfigurationInfo) func(w http.ResponseWriter, r *http.Request) {
-	return func(w http.ResponseWriter, r *http.Request) {
-		if corsInfo.EnableCORS && r.Header.Get(Origin) != "" {
-			if len(corsInfo.CORSAllowedMethods) > 0 {
-				w.Header().Set(AccessControlAllowMethods, corsInfo.CORSAllowedMethods)
+func HandlePreflight(corsInfo config.CORSConfigurationInfo) echo.MiddlewareFunc {
+	return func(next echo.HandlerFunc) echo.HandlerFunc {
+		return func(c echo.Context) error {
+			r := c.Request()
+			// skip the middleware if AccessControlRequestMethod header is not defined
+			if r.Header.Get(AccessControlRequestMethod) == "" {
```

Comment:
How would `AccessControlRequestMethod` be set?
Is this a bug in 3.0/2.x ?

Reply:
@bnevis-i , so this change is also fixing a bug in 3.0 & 2.x, which should be documented.

Reply:
No bug.  It's there in the previous release, but in httpserver.go

```
        b.router.Methods(http.MethodOptions).MatcherFunc(func(r *http.Request, rm *mux.RouteMatch) bool {
                return r.Header.Get(AccessControlRequestMethod) != ""
        }).HandlerFunc(HandlePreflight(bootstrapConfig.Service.CORSConfiguration))
```